### ISSUE TITLE
difference in boundary output between polygon and linestring as footprint input

### ISSUE THREAD

**Issue by tomvantilburg** [Geodan/building-boundary#16]
It is expected that when converting the polygon to a linestring, the effect on the boundary directions is the same. According to a first test, it is not:

Using no footprint:
![image](https://user-images.githubusercontent.com/1998463/60332707-b195b280-9997-11e9-867d-fa049b567e27.png)

Using linestring (exterior-ring of polygon):
![image](https://user-images.githubusercontent.com/1998463/60332883-fd485c00-9997-11e9-9963-8370d58c82f9.png)

Using polygon:
![image](https://user-images.githubusercontent.com/1998463/60332200-9c6c5400-9996-11e9-9490-89a31ab908ea.png)

**Comment by chrislcs** [comment]
Should be fixed by 2a0d4cf

**Comment by chrislcs** [comment]
Does this fix the issue? 4c1e67b
If not, I could use an example to debug

**Comment by tomvantilburg** [comment]
Still subtle differences between using a linestring and a polygon. See example on metis in: 
`/home/tomt/notebooks/makeroofs/Pyntcloud ransac.ipynb`

**Comment by tomvantilburg** [comment]
Reproduce with: 
```
import numpy as np
import pandas as pd
import psycopg2 

conn = psycopg2.connect("host='{}' port={} dbname='{}' user={} password={}".format('localhost', 5433, 'maquette_nl', 'chrisl', 'smurf'))
sql = "select x,y,z from chrisl.points;"
points = pd.read_sql_query(sql, conn)
conn = None



conn = psycopg2.connect("dbname=maquette_nl port=5433")
cur = conn.cursor()
cur.execute("""SELECT ST_AsText(ST_Force2D(geom)) as geom  FROM work.blocks WHERE blockid = '0363100012135614'""")
row = cur.fetchone()
cur.close()
conn.close()
polygonwkt=row[0]


conn = psycopg2.connect("dbname=maquette_nl port=5433")
cur = conn.cursor()
cur.execute("""
WITH segments AS (
    SELECT blockid, ST_MakeLine(lag((pt).geom, 1, NULL) OVER (PARTITION BY blockid, (pt).path[1] ORDER BY (pt).path), (pt).geom) AS geom
    FROM (SELECT blockid, ST_DumpPoints(ST_Force2D(geom)) AS pt FROM work.blocks WHERE blockid = '0363100012135614') as dumps
)
SELECT ST_AsText(ST_Force2D(ST_Collect(geom))) as geom  FROM segments
--WHERE ST_Length(geom) > 2
;
""")
row = cur.fetchone()
cur.close()
conn.close()
segmentswkt=row[0]

from buildingboundary import trace_boundary
from shapely.geometry import Polygon

pts = np.array((points['x'],points['y'])).T

vertices1 = trace_boundary(pts,
    max_error = 0.26,
    merge_angle = 0.01,
    k=5,
    alpha=0.5,
    min_area=4,
    max_merge_distance=0.6,
    max_error_invalid=0.35,
    inflate=True,
    footprint_geom=None
)

vertices2 = trace_boundary(pts,
    max_error = 0.26,
    merge_angle = 0.01,
    k=5,
    alpha=0.5,
    min_area=4,
    max_merge_distance=0.6,
    max_error_invalid=0.35,
    inflate=True,
    footprint_geom=polygonwkt
)

vertices3 = trace_boundary(pts,
    max_error = 0.26,
    merge_angle = 0.01,
    k=5,
    alpha=0.5,
    min_area=4,
    max_merge_distance=0.6,
    max_error_invalid=0.35,
    inflate=True,
    footprint_geom=segmentswkt
)

import matplotlib.pyplot as plt
from matplotlib.patches import Polygon

fig = plt.figure(figsize=(30, 30))
sub=plt.subplot(311,aspect='equal')
plt.scatter(points['x'],points['y'],
            c='red',
            cmap='Paired', s=12)
polygon_geom = Polygon(vertices1, fill=False, hatch='/')
sub.add_patch(polygon_geom)
    
sub=plt.subplot(312,aspect='equal')
plt.scatter(points['x'],points['y'],
            c='red',
            cmap='Paired', s=12)
polygon_geom = Polygon(vertices2, fill=False, hatch='/')
sub.add_patch(polygon_geom)

sub=plt.subplot(313,aspect='equal')
plt.scatter(points['x'],points['y'],
            c='red',
            cmap='Paired', s=12)
polygon_geom = Polygon(vertices3, fill=False, hatch='/')
sub.add_patch(polygon_geom)
```

**Comment by chrislcs** [comment]
Fixed a bug in 567a6fd. Small differences can still occur due to the random nature of ransac. If you set the `random_state` of the ransac function to a fixed value the results should be identical.

```python
ransac(points, LineModelND,
       min_samples=2,
       residual_threshold=distance,
       max_trials=1000,
       random_state=1)
```
